src/controller/ApplicationController.tsx

```diff
@@ -128,12 +130,21 @@ class ApplicationController<
         }`
       );
 
-      // @ts-ignore
+      // @ts-ignore props are readonly, as we don't want them reassigned, but we need to set them here
+      this.props = store({ ...initialArgs });
+
       this.state = store(cloneDeep(this.initialState));
       if (this.initialize) this.initialize(initialArgs);
       this.initialized = true;
     } else {
-      this.changeProps(initialArgs);
+      const oldProps = { ...this.props };
+      Object.keys(initialArgs).forEach(key => {
+        if (this.props[key] !== initialArgs[key]) {
+          this.props[key] = initialArgs[key];
+        }
+      });
+
+      this.changeProps(initialArgs, oldProps);
```

Comment:
Moved this down, because I think it's better for `this.props` to be current within this method, even though the new props are passed as the first argument.